UM/Qt/qml/UM/MessageStack.qml

```diff
@@ -91,16 +92,122 @@ ListView
                 margins: UM.Theme.getSize("default_margin").width
             }
 
-            height: childrenRect.height
+            height: Math.max(messageTypeIcon.height, messageTitle.height)
+            Item
+            {
+                id: messageTypeIcon
+                visible: messageIcon.source != ""
+                height: UM.Theme.getSize("small_button_icon").height
+                width: visible ? UM.Theme.getSize("small_button_icon").height : 0
+                anchors.verticalCenter: parent.verticalCenter
+                Rectangle
+                {
+                    id: messageIconBackground
+                    height: parent.height
+                    width: parent.width
+                    radius: Math.round(width / 2)
+                }
+                UM.RecolorImage
+                {
+                    id: messageIcon
+                    height: parent.height
+                    width: parent.width
+                    sourceSize.width: width
+                    sourceSize.height: height
+                }
+
+                states:
+                [
+                    State
+                    {
+                        name: "positive"
+                        when: model.message_type == 0
+                        PropertyChanges
+                        {
+                            target: messageIcon
+                            source: UM.Theme.getIcon("Check", "low")
+                            color: UM.Theme.getColor("message_success_icon")
+                        }
+                        PropertyChanges
+                        {
+                            target: messageIconBackground
+                            color: UM.Theme.getColor("message_success_icon_background")
```

Comment:
Rather than hard-coding specific theme entries for specific details in the interface, the design philosophy is to have re-usable theme entries as much as possible.
A better theme entry could be something like `success_contrast` (to indicate that it should contrast with the "success" theme entry) or `success_background`. Although in this case it appears more like a foreground than a background.

Reply:
There was no success color so I added it as a generic one. The message is also using that now.